File: Programmers/Level 2/카펫.js
function getDivisor(num) {
  const divisors = [];

  for (let i = 1; i <= Math.sqrt(num); i++) {
    if (num % i === 0) divisors.push([num / i, i]);
  }

  return divisors;
}

function solution(brown, yellow) {
  const divisorBY = getDivisor(brown + yellow);
  const divisorY = getDivisor(yellow);

  for (const [M, N] of divisorBY) {
    for (const [a, b] of divisorY) {
      if (a + 2 <= M && b + 2 <= N) return [M, N];
    }
  }
}

console.log(solution(10, 2));
console.log(solution(8, 1));
console.log(solution(24, 24));

File: Python/2Level/카펫.py
import math


def solution(brown, yellow):
    totalCells = brown + yellow

    until = int(math.floor(math.sqrt(yellow))) + 1

    for x in range(1, until):
        if yellow % x != 0:
            continue

        y = int(yellow / x)

        if totalCells == (x + 2) * (y + 2):
            return [y + 2, x + 2]


# print(solution(10, 2))
# print(solution(8, 1))
print(solution(24, 24))
